src/psyclone/psyir/nodes/omp_directives.py

```diff
@@ -169,41 +172,37 @@ def validate_global_constraints(self):
                 "OMPSingleDirective must be inside an OMP parallel region but "
                 "could not find an ancestor OMPParallelDirective node")
 
```

Comment:
Apologies but could you put this check back? (It may be possible for a user to construct an invalid tree without using the transformation and we still need to catch that at code-generation time.)

Reply:
I can do but I have no idea how to create a test for it (since my only idea was using the transformation) - is there a similar test I can base it on?

Reply:
Added this back in - will try to test it now

Reply:
Done

Reply:
Thanks.